xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -286,55 +270,83 @@ def match_and_rewrite(self, op: AccessOp, rewriter: PatternRewriter, /):
         rewriter.replace_matched_op([*off_const_ops, *off_sum_ops, load], [load.res])
 
 
-@dataclass
 class StencilTypeConversionFuncOp(RewritePattern):
-    return_targets: dict[ReturnOp, list[CastOp | memref.Subview | None]]
-
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: FuncOp, rewriter: PatternRewriter, /):
         inputs: list[Attribute] = []
         for arg in op.body.block.args:
             if isa(arg.typ, FieldType[Attribute]):
-                memreftyp = GetMemRefFromField(arg.typ)
+                memreftyp = StencilToMemRefType(arg.typ)
                 rewriter.modify_block_argument_type(arg, memreftyp)
                 inputs.append(memreftyp)
             else:
                 inputs.append(arg.typ)
+        outputs: list[Attribute] = [
+            StencilToMemRefType(out) if isa(out, FieldType[Attribute]) else out
+            for out in op.function_type.outputs
+        ]
+        op.attributes["function_type"] = FunctionType.from_lists(inputs, outputs)
 
-        op.attributes["function_type"] = FunctionType.from_lists(
-            inputs, list(op.function_type.outputs.data)
-        )
 
+class TypeSpreadingForOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: scf.For, rewriter: PatternRewriter, /):
+        for i in range(len(op.iter_args)):
+            arg_typ = op.body.block.args[i + 1].typ
+            iter_typ = op.iter_args[i].typ
+            if arg_typ != iter_typ:
+                rewriter.modify_block_argument_type(
+                    op.body.block.args[i + 1], op.iter_args[i].typ
+                )
+                for y in [y for y in op.body.ops if isinstance(y, scf.Yield)]:
```

Comment:
Should just be `body.last_op` or something, no need to iterate. Yield will always be the last op, that's the verifiers job.

And there should always only be one.